kernel/kernel-api/src/main/java/io/delta/kernel/internal/tablefeatures/TableFeatures.java

```diff
@@ -516,10 +506,10 @@ public static boolean isDomainMetadataSupported(Protocol protocol) {
   /// Private methods                                                           ///
   /////////////////////////////////////////////////////////////////////////////////
   /**
-   * Extracts all new table features (and their dependency features) that are supported by the given
-   * metadata. A feature is considered new if it is not already supported by the current protocol.
+   * Extracts all table features (and their dependency features) that are enabled by the given
+   * metadata and supported in existing protocol.
    */
-  private static Set<TableFeature> extractAutomaticallyEnabledNewFeatures(
+  private static Set<TableFeature> extractAllNeededTableFeatures(
       Metadata newMetadata, Protocol currentProtocol) {
```

Comment:
need to update line 528 to not remove the features?

Reply:
done